validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/KeystoreLockerTest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import com.google.common.io.Resources;
+import java.nio.file.Path;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+
+public class KeystoreLockerTest {
+
+  private KeystoreLocker keystoreLocker = new KeystoreLocker();
+
+  @Test
+  void shouldLockKeystoreFileAndFailWhenTryingCreateLockForLockedFile() throws Exception {
```

Comment:
We need to add tests for existing PID files that have PIDs for live processes (will need to spawn something via `ProcessBuilder` - maybe run `/bin/sleep` and skip the test if it can't be executed like on Windows).  For PID files that have a PID that isn't alive, empty PID files and PID files containing something that isn't a long.

And probably also a non-existent directory to trigger an IOException while creating the file.

Reply:
I've added all the tests but the last one you mentioned with the non-existent directory. Not sure how I could mock a scenario like that.

Reply:
Create a temp dir (which does exist) then ask `KeystoreLocker` to lock a non-existent keystore in a non-existent sub directory.

Reply:
Done.